File: CSharp/Behavioral/Mediator/Example2/Show/Show/NonTerminalExpression.cs
namespace Show
{
    public class NonTerminalExpression: Expression
    {
        private Expression expr1;
        private Expression expr2;

        public NonTerminalExpression(Expression expr1, Expression expr2) {
            this.expr1 = expr1;
            this.expr2 = expr2;
        }
        
        public bool Interpret(string context)
        {
            return expr1.Interpret(context) && expr2.Interpret(context);
        }
    }
}

File: TheRayCode/JavaScript/Behavioral/Interpreter/Example1/NonterminalExpression.js
// NonterminalExpression.js

const AbstractExpression = require('./AbstractExpression');

// NonterminalExpression: combines two expressions
class NonterminalExpression extends AbstractExpression {
    constructor(expr1, expr2) {
        super();
        this.expr1 = expr1; // left side
        this.expr2 = expr2; // right side
    }

    interpret(context) {
        // for simplicity, logical AND of both expressions
        return this.expr1.interpret(context) && this.expr2.interpret(context);
    }
}

module.exports = NonterminalExpression;
